sycl/include/sycl/reduction.hpp

```diff
@@ -1147,11 +1168,11 @@ template <> struct NDRangeReduction<reduction::strategy::range_basic> {
         // Copy the element to local memory to prepare it for tree-reduction.
         LocalReds[LID] = getReducerAccess(Reducer).getElement(E);
 
-        doTreeReduction(WGSize, LID, false, Identity, LocalReds, BOp,
+        doTreeReduction(WGSize, LID, LocalReds, BOp,
                         [&]() { workGroupBarrier(); });
```

Comment:
Unrelated, but do we really need the lambda? Does `workGroupBarrier` have defaulted parameters that we "specify" by using the lambda?

Reply:
Without the lambda the compiler will complain about not being able to call through a function pointer.